src/js/common/schemaform/FileField.jsx

```diff
@@ -0,0 +1,161 @@
+import PropTypes from 'prop-types';
+import React from 'react';
+import _ from 'lodash/fp';
+// import Scroll from 'react-scroll';
+import { displayFileSize } from '../../common/utils/helpers';
+import environment from '../../common/helpers/environment';
+
+export default class FileField extends React.Component {
+  onAddFile = (e) => {
+    const files = this.props.formData || [];
+    this.props.onChange(files.concat({
+      uploading: true
+    }));
+    this.uploadFile(e.target.files[0])
+      .then(fileInfo => {
+        const uploadingFiles = this.props.formData;
+        this.props.onChange(_.set(uploadingFiles.length - 1, {
+          errorMessage: null,
+          uploading: false,
+          fileName: fileInfo.fileName,
+          maxSize: fileInfo.maxSize,
+          confirmationNumber: fileInfo.confirmationNumber
+        }, uploadingFiles));
+      })
+      .catch(message => {
+        const uploadingFiles = this.props.formData || [];
```

Comment:
Yeah, there's an `onChange` call before the promise, so this will have re-rendered by the time the promise is resolved.